plugins/debuginfo-install.py

```diff
@@ -74,59 +76,170 @@ def configure(self):
         self.base.repos.enable_debug_repos()
 
     def run(self):
-        self.packages = self.base.sack.query()
-        self.packages_available = self.packages.available()
         errors_spec = []
 
+        debuginfo_suffix_len = len(Package.DEBUGINFO_SUFFIX)
+        debugsource_suffix_len = len(Package.DEBUGSOURCE_SUFFIX)
+
         for pkgspec in self.opts.package:
-            package_query = sorted(dnf.subject.Subject(pkgspec).get_best_query(
-                self.cli.base.sack).filter(arch__neq='src'), reverse=True)
-            if not package_query:
-                msg = _('No match for argument: %s')
-                logger.info(msg, self.base.output.term.bold(pkgspec))
+            solution = dnf.subject.Subject(pkgspec).get_best_solution(self.base.sack)
+
+            query = solution["query"].filterm(arch__neq='src')
+
+            package_dict = query.available()._name_dict()
+            # installed versions of packages have priority, replace / add them to the dict
+            package_dict.update(query.installed()._name_dict())
+
+            if not package_dict:
```

Comment:
Please could you move the code 
```

            if not query:
                logger.info(_('No match for argument: %s'), self.base.output.term.bold(pkgspec))
                errors_spec.append(pkgspec)
                continue
```
before creation of dicts. It is more efficient and transparent.

Reply:
Will do.